### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by maksym-nemera**

### CODE REVIEW

File: src/makeRobotAccountant.js [hunk 1]
```diff
@@ -6,7 +6,29 @@
...
-  // write code here
+  let count = 0;
+
+  return function(x) {
+    return function(y) {
+      const sum = x + y;
+
+      count++;
+
+      if (count === 4) {
```
Comment (dinaminahi) [inline src/makeRobotAccountant.js]: Redundant check, the bottom one is doing the same

File: src/makeRobotAccountant.js [hunk 2]
```diff
@@ -6,7 +6,29 @@
...
+        return 'Bzzz... Error!';
+      }
+
+      if (count > 3) {
+        if (count % 2 === 0) {
+          return 'Bzzz... Error!';
+        }
+
+        return sum;
+      }
```
Comment (dinaminahi) [inline src/makeRobotAccountant.js]:
```suggestion
      if (count > 3 && count % 2 === 0) {
          return 'Bzzz... Error!';
      }
```

**Review by dinaminahi — CHANGES_REQUESTED** [review]

File: src/makeRobotAccountant.js [hunk 3]
```diff
@@ -6,7 +6,21 @@
  */
 
 function makeRobotAccountant() {
-  // write code here
+  let count = 0;
+
+  return function(x) {
```
Comment (33Orange) [inline src/makeRobotAccountant.js]: Rename the function parameter 'x' to a more descriptive name, such as 'num1'.

File: src/makeRobotAccountant.js [hunk 4]
```diff
@@ -6,7 +6,21 @@
...
+    return function(y) {
```
Comment (33Orange) [inline src/makeRobotAccountant.js]: Rename the function parameter 'y' to a more descriptive name, such as 'num2'.

File: src/makeRobotAccountant.js [hunk 5]
```diff
@@ -6,7 +6,21 @@
...
+      const sum = x + y;
```
Comment (33Orange) [inline src/makeRobotAccountant.js]: Move the 'const sum' declaration to after the if statement, so it is only calculated when needed.
You can immediately return the amount without adding a variable here.

**Review by 33Orange — CHANGES_REQUESTED** [review]
Some comments:

**Review by dinaminahi — APPROVED** [review]
Well done 💪

**Pull request closed without merging**